src/entities/trades/swapr-v3/SwaprV3.ts

```diff
@@ -0,0 +1,235 @@
+import type { BaseProvider } from '@ethersproject/providers'
+import dayjs from 'dayjs'
+
+import { Currency as CurrencyType, Fraction, validateAndParseAddress } from '@uniswap/sdk-core'
+import invariant from 'tiny-invariant'
+
+import { CurrencyAmount, Percent, Price, TokenAmount } from '../../fractions'
+
+import { TradeWithSwapTransaction } from '../interfaces/trade'
+import { RoutablePlatform } from '../routable-platform'
+import { getProvider, tryGetChainId } from '../utils'
+import { ChainId, ONE, TradeType } from '../../../constants'
+
+import { UnsignedTransaction } from 'ethers'
+import { TradeOptions } from '../interfaces/trade-options'
+import { parseUnits } from '@ethersproject/units'
+import { Route } from './entities/route'
+import { SWAPR_ALGEBRA_CONTRACTS } from './constants'
+import { getQuoterContract, getRouterContract } from './contracts'
+import { getRoutes } from './routes'
+import { maximumSlippage as defaultMaximumSlippage } from '../constants'
+import { Currency } from '../../currency'
+import { Token, WXDAI } from '../../token'
+
+interface SwaprV3ConstructorParams {
+  maximumSlippage: Percent
+  inputAmount: CurrencyAmount
+  outputAmount: CurrencyAmount
+  tradeType: TradeType
+  chainId: number
+  priceImpact: Percent
+  fee: Percent
+}
+
+export interface SwaprV3GetQuoteParams {
+  amount: CurrencyAmount
+  quoteCurrency: Currency
+  tradeType: TradeType
+  maximumSlippage?: Percent
+  recipient?: string
+}
+
+export class SwaprV3Trade extends TradeWithSwapTransaction {
+  public constructor({
+    inputAmount,
+    outputAmount,
+    maximumSlippage,
+    priceImpact,
+    tradeType,
+    chainId,
+    fee,
+  }: SwaprV3ConstructorParams) {
+    super({
+      details: undefined,
+      type: tradeType,
+      inputAmount,
+      outputAmount,
+      maximumSlippage,
+      platform: RoutablePlatform.SWAPR_V3,
+      chainId,
+      executionPrice: new Price({
+        baseCurrency: inputAmount.currency,
+        quoteCurrency: outputAmount.currency,
+        denominator: inputAmount.raw,
+        numerator: outputAmount.raw,
+      }),
+      priceImpact,
+      fee,
+      approveAddress: SWAPR_ALGEBRA_CONTRACTS['router'],
+    })
+  }
+
+  static async getQuote(
+    { amount, quoteCurrency, tradeType, maximumSlippage }: SwaprV3GetQuoteParams,
+    provider?: BaseProvider
+  ): Promise<SwaprV3Trade | null> {
+    const chainId = tryGetChainId(amount, quoteCurrency)
+    invariant(chainId, 'SwaprV3Trade.getQuote: chainId is required')
+
+    maximumSlippage = maximumSlippage ?? defaultMaximumSlippage
+    provider = provider ?? getProvider(chainId)
+
+    const tokenIn = Currency.isNative(amount.currency)
+      ? WXDAI[ChainId.GNOSIS]
+      : new Token(
+          ChainId.GNOSIS,
+          // @ts-expect-error
```

Comment:
Why do we have this commented?

Reply:
We need it otherwise it mismatch the types, as the algebra code uses Token from Uniswap sdk. Didn't find a cleaner solution.